### PR TITLE
Enable `doNotStrip` when doing development/debugging within Android Studio

### PR DESCRIPTION
**Pull request by m4gr3d**
The updated logic automatically detect when running within Android Studio, and enable `doNotStrip` in order to keep around the debug symbols.

### CODE REVIEW

**Comment by m4gr3d** [comment]
cc @thebestnom 

For your update, you may want to update the `shouldNotStrip` function in order to check for the property passed on the command line.

**Comment by BastiaanOlij** [comment]
So far so good, it nicely hits breakpoints where I set them.
It did show something in the log about there not being symbols though so this may work simply because we're running from withing Android Studio and Android Studio thus has access to the source. 
Unfortunately the log disappeared before I could grab it and I can't find where it's hidden now :)

**Comment by BastiaanOlij** [comment]
One thing I think we should add, not 100% related to this, but is adding a `.gitignore` entry so our assets folder is there but its contents is not added to git.

**Comment by m4gr3d** [comment]
> One thing I think we should add, not 100% related to this, but is adding a `.gitignore` entry so our assets folder is there but its contents is not added to git.

@BastiaanOlij I've added the `.gitignore` entry in the `assets` directory.

**Comment by m4gr3d** [comment]
@akien-mga The second commit addresses the comment made in https://github.com/godotengine/godot/pull/42980/files/37d515e99c2da449397511cbc9b25fee84dd66fc#diff-d06617966387fb7135d4af8cf52217b94e7a4c658c6ed9f259bbf8cd801e31d5.

Let me know if it should still be squashed into a single commit.

**Comment by BastiaanOlij** [comment]
@m4gr3d mentioned in chat as well but just to have it on the PR. If the binary project file does not exists shouldn't `err != ERR_FILE_NOT_FOUND` be enough to ignore the error and load the text version of the file? I still think the change is valid because if the file can't be read due to other reasons it would be good to just report it and then attempt to load the text version as a fallback especially in the editor.

But sounds to me like it's giving the wrong error on a missing file?

**Comment by akien-mga** [comment]
@m4gr3d The second commit should have been in its own PR, it's not related to enabling `doNotStrip` on Android builds in any way.

For the Android `doNotStrip` change I'm waiting for a decision from the Android team on what to merge as there's also #46880 which seems to build on top of it, it's not sure which should be merged.

The other change is a `core` fix which is not linked to Android buildsystem.

**Comment by BastiaanOlij** [comment]
@akien-mga I'd merge this (after @m4gr3d moves the second change into its own PR) and then ask @thebestnom to rebase his PR. 
They are related but different :)

The second fix from Fredia is pretty important to run Godot directly from Android Studio (I think I got stuck on this as well)

**Comment by m4gr3d** [comment]
> @m4gr3d The second commit should have been in its own PR, it's not related to enabling `doNotStrip` on Android builds in any way.
> 
> For the Android `doNotStrip` change I'm waiting for a decision from the Android team on what to merge as there's also #46880 which seems to build on top of it, it's not sure which should be merged.
> 
> The other change is a `core` fix which is not linked to Android buildsystem.

@akien-mga The second commit is related because the first commit enables to run Godot projects from Android studio (I'm doing a write up describing the process to be soon added to the Android docs), but the issue resolved by the second commit prevents that from working. So the first commit by itself is broken by the issue addressed in the second commit.
It can be split if you still think it should.

PR #46880 builds on top of this, so like @BastiaanOlij mentioned, this should be merged so that @thebestnom can rebase #46880.

**Comment by akien-mga** [comment]
> The second commit is related because the first commit enables to run Godot projects from Android studio (I'm doing a write up describing the process to be soon added to the Android docs), but the issue resolved by the second commit prevents that from working. So the first commit by itself is broken by the issue addressed in the second commit.

Alright, but the second commit is still fixing an unrelated issue which does happen to affect Android Studio debugging, but would also affect running any project with a similar situation, i.e.:
- `project.binary` exists but can't be opened
- `project.godot` exists and you want to fall back to it instead of aborting

The fact that `project.binary` exists but can't be opened is a bug and should likely be investigated as such. This commit works it around (bug still present, but doesn't prevent starting the game), but it should still be debugged, and a dedicated PR would be a cleaner way to have this reviewed by people familiar with this very core aspect of the engine's code (used by all platforms, not just Android).

**Comment by m4gr3d** [comment]
I removed the second commit and create a separate PR for it: https://github.com/godotengine/godot/pull/47065

> 
> Alright, but the second commit is still fixing an unrelated issue which does happen to affect Android Studio debugging, but would also affect running any project with a similar situation, i.e.:
> 
> * `project.binary` exists but can't be opened

This is not an issue (at least as far as I'm aware). I described the use-case where we encountered the issue in https://github.com/godotengine/godot/pull/42980#discussion_r594759905_. Since it's a development use-case, the `project.binary` is not generated and we rely on `project.godot`.
A possible issue would be why `ERR_CANT_OPEN` is returned when the file is not present instead of `ERR_FILE_NOT_FOUND`, but regardless I think the fallback fix is still worth having.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.

**Pull request merged**
